src/auth/jwtclient.ts

```diff
@@ -148,10 +152,22 @@ export class JWT extends OAuth2Client implements IdTokenProvider {
             this.eagerRefreshThresholdMillis
           );
         }
+
+        let scopes: string | string[] | undefined;
+        if (this.hasUserScopes()) {
+          scopes = this.scopes;
+        } else if (!url) {
+          scopes = this.defaultScopes;
+        }
+
         const headers = await this.access.getRequestHeaders(
-          url,
-          this.additionalClaims
+          url ?? undefined,
```

Comment:
NodeJS question: why isn't it enough to just pass in the falsy value (e.g. `url` without the `?? undefined` check)? Is it for cases where it's `false` or an empty string?

Reply:
`null` and `false` are considered valid arguments for an optional string parameter, which can only be a string or `undefined`.